packages/server/lib/browsers/index.ts

```diff
@@ -204,8 +206,11 @@ export = {
     // this browser, it means it has been orphaned and should be terminated.
     //
     // https://github.com/cypress-io/cypress/issues/24377
-    if (thisLaunchAttempt !== launchAttempt) {
-      await kill({ instance: _instance, nullOut: false })
+    const isOrphanedBrowserProcess = thisLaunchAttempt !== launchAttempt
+
+    if (isOrphanedBrowserProcess) {
+      debug(`killing process because launch attempt: ${thisLaunchAttempt} does not match current launch attempt: ${launchAttempt}`)
+      await kill({ instance: _instance, isOrphanedBrowserProcess, nullOut: false })
```

Comment:
Is nullOut (or rather !nullOut) serving the same purpose as isOrphanedBrowserProcess?

Reply:
I don't think so. I think the goal of `nullOut` is to set the current browser instance to null, which we don't want to do since we want the active instance to be set, but the `_instance` inside the closer to be cleaned up.